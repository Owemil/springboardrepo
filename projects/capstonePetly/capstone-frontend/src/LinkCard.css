.card-link {
    margin-left: 1rem;
    padding-top: .5rem;
}

.card-link-meta {
    margin-left: 1rem;

}

.card-link :hover {
    cursor: pointer;
    box-shadow: 1px 1px 6px black;
}

.lc-div {
    padding-left: 4rem;
    display: inline-block;
}

.others-toast {
    position: absolute;
    z-index: 99;
}