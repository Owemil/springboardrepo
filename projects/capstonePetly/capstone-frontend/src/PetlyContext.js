import React from "react";

const PetlyContext = React.createContext();

export default PetlyContext;