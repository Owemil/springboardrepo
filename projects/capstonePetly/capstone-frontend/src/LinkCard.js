import React from "react";
import "./LinkCard.css"
import { useNavigate } from "react-router-dom";
import { Card, CardBody, CardText, CardFooter, CloseButton, Toast, ToastHeader, ToastBody } from "reactstrap";
import useToggle from "./helpers/useToggle"

// LinkCard component takes in meta data from api requests to inform on animal amount and to link to search pages

function LinkCard({ meta, search }) {
    const [isOpen, toggle] = useToggle()
    const other = ["Birds", "Rabbits", "Horses", "Barnyard", "Reptile", "Small & Furry", "Scales, Shells & Other"]
    const navigate = useNavigate()

    if (meta) {
        return (
            <>
                <Card
                    style={{
                        width: '18rem'
                    }}
                    className="card-link-meta"
                    onClick={() => { return navigate(`/search`) }}
                >

                    <CardBody className='link-body'>
                        <CardText>
                            View {meta.count} more pets nearby!
                        </CardText>
                        <CardFooter className="lc-footer">Meet Them!</CardFooter>
                    </CardBody>
                </Card>

            </>
        )
    } else if (search === "Other Animals") {
        return (
            <div>
                <Card
                    style={{
                        width: '8.5rem',
                        height: '5rem'
                    }}
                    className="card-link"
                    onClick={toggle}
                >

                    <CardBody className='link-body'>
                        <CardText>
                            {search}
                        </CardText>
                    </CardBody>
                </Card>

                <Toast isOpen={isOpen} className="others-toast">
                    <ToastHeader >
                        <CloseButton onClick={toggle} />
                    </ToastHeader>
                    <ToastBody>
                        <p>What type of pet are you searching for?</p>
                        <div>
                            {other.map((o, idx) => {
                                return (
                                    <Card
                                        onClick={() => {
                                            return navigate(`/search/${o}`)
                                        }}
                                        key={idx}
                                    >

                                        <CardBody>
                                            <CardText>
                                                {o}
                                            </CardText>
                                        </CardBody>
                                    </Card>
                                )
                            })}

                        </div>
                    </ToastBody>
                </Toast>
            </div>
        )
    } else if (search === "Organizations") {
        return (
            <Card
                style={{
                    width: '8.5rem',
                    height: '5rem'
                }}
                className="card-link"
                onClick={() => { return navigate(`/orgs`) }}
            >

                <CardBody className='link-body'>
                    <CardText>
                        {search}
                    </CardText>
                </CardBody>
            </Card>
        )
    } else {
        return (
            <Card
                style={{
                    width: '8.5rem',
                    height: '5rem'
                }}
                className="card-link"
                onClick={() => { return navigate(`/search/${search}`) }}
            >

                <CardBody className='link-body'>
                    <CardText>
                        {search}
                    </CardText>
                </CardBody>
            </Card>
        )
    }

}



export default LinkCard




