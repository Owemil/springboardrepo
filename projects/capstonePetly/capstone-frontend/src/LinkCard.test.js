import React from "react";
import PetlyContext from "./PetlyContext.js";
import { fireEvent, render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import LinkCard from "./LinkCard.js"

const meta = { count: 248 }

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: () => mockedUsedNavigate,
}));

it("should render (meta)", function () {

    render(
        <MemoryRouter>
            <LinkCard meta={meta} />
        </MemoryRouter>)
})
it("should render (search-Cats)", function () {

    render(
        <MemoryRouter>
            <LinkCard search="Cats" />
        </MemoryRouter>)
})
it("should render (search-Dogs)", function () {
    render(
        <MemoryRouter>
            <LinkCard search="Dogs" />
        </MemoryRouter>)
})
it("should render (search-OtherAnimals)", function () {
    render(
        <MemoryRouter>
            <LinkCard search="Other Animals" />
        </MemoryRouter>)
})
it("should render (search-Organizations)", function () {
    render(
        <MemoryRouter>
            <LinkCard search="Organizations" />
        </MemoryRouter>)
})

it("should match snapshot (meta)", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LinkCard meta={meta} />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot (search-Cats)", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LinkCard search="Cats" />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot (search-Dogs)", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LinkCard search="Dogs" />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot (search-OtherAnimals)", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LinkCard search="Other Animals" />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})
it("should match snapshot (search-Organizations)", function () {
    const { asFragment } = render(
        <MemoryRouter>
            <LinkCard search="Organizations" />
        </MemoryRouter>)

    expect(asFragment).toMatchSnapshot()
})

it("should navigate (meta)", function () {
    const { getByText } = render(
        <MemoryRouter>
            <LinkCard meta={meta} />
        </MemoryRouter>
    )

    const card = getByText(/nearby/)

    fireEvent.click(card)

    expect(mockedUsedNavigate).toHaveBeenCalled()
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/search')
})
it("should navigate (search-Cats)", function () {
    const { getByText } = render(
        <MemoryRouter>
            <LinkCard search="Cats" />
        </MemoryRouter>
    )

    const card = getByText("Cats")

    fireEvent.click(card)

    expect(mockedUsedNavigate).toHaveBeenCalled()
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/search/Cats')
})
it("should navigate (search-Dogs)", function () {
    const { getByText } = render(
        <MemoryRouter>
            <LinkCard search="Dogs" />
        </MemoryRouter>
    )

    const card = getByText("Dogs")

    fireEvent.click(card)

    expect(mockedUsedNavigate).toHaveBeenCalled()
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/search/Dogs')
})
it("should navigate (search-OtherAnimals)", function () {
    const { getByText } = render(
        <MemoryRouter>
            <LinkCard search="Other Animals" />
        </MemoryRouter>
    )

    const card = getByText("Other Animals")

    fireEvent.click(card)

    const rabbitCard = getByText("Rabbits")

    fireEvent.click(rabbitCard)

    expect(mockedUsedNavigate).toHaveBeenCalled()
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/search/Rabbits')
})
it("should navigate (search-Organizations)", function () {
    const { getByText } = render(
        <MemoryRouter>
            <LinkCard search="Organizations" />
        </MemoryRouter>
    )

    const card = getByText("Organizations")

    fireEvent.click(card)

    expect(mockedUsedNavigate).toHaveBeenCalled()
    expect(mockedUsedNavigate).toHaveBeenCalledWith('/orgs')
})