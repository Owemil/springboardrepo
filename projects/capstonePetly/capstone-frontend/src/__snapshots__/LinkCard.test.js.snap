// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot (meta) 1`] = `[Function]`;

exports[`should match snapshot (search-Cats) 1`] = `[Function]`;

exports[`should match snapshot (search-Dogs) 1`] = `[Function]`;

exports[`should match snapshot (search-Organizations) 1`] = `[Function]`;

exports[`should match snapshot (search-OtherAnimals) 1`] = `[Function]`;
